ethcore/src/client/client.rs

```diff
@@ -1338,27 +1338,41 @@ impl BlockChainReset for Client {
 		for hash in &blocks_to_delete {
 			db_transaction.delete(::db::COL_HEADERS, &hash.hash());
 			db_transaction.delete(::db::COL_BODIES, &hash.hash());
-			db_transaction.delete(::db::COL_EXTRA, &hash.hash());
+			Writable::delete::<BlockDetails, H264>(&mut db_transaction, ::db::COL_EXTRA, &hash.hash());
```

Comment:
Shouldn't it be `Writable::delete::<H264, BlockDetails>`, no?

Reply:
That's not how, the `Key` trait works, yeah, it confused me too.

https://github.com/paritytech/parity-ethereum/blob/7c335e87642b848f61ab3ee7bbff5964b53d6f11/ethcore/db/src/db.rs#L95-L98

https://github.com/paritytech/parity-ethereum/blob/7c335e87642b848f61ab3ee7bbff5964b53d6f11/ethcore/db/src/keys.rs#L82-L88

https://github.com/paritytech/parity-ethereum/blob/7c335e87642b848f61ab3ee7bbff5964b53d6f11/ethcore/db/src/db.rs#L83-L89